src/anomalib/metrics/per_image/binclf_curve_numpy.py

```diff
@@ -0,0 +1,430 @@
+"""Binary classification curve (numpy-only implementation).
+
+A binary classification (binclf) matrix (TP, FP, FN, TN) is evaluated at multiple thresholds.
+
+The thresholds are shared by all instances/images, but their binclf are computed independently for each instance/image.
+"""
+
+# Original Code
+# https://github.com/jpcbertoldo/aupimo
+#
+# Modified
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import itertools
+import logging
+from enum import Enum
+from functools import partial
+
+import numpy as np
+from numpy import ndarray
+
+try:
+    import numba  # noqa: F401
+except ImportError:
+    HAS_NUMBA = False
+else:
+    HAS_NUMBA = True
+
+
+if HAS_NUMBA:
+    from . import _binclf_curve_numba
+
+from . import _validate
+
+logger = logging.getLogger(__name__)
+
+# =========================================== CONSTANTS ===========================================
+
+
+class BinclfAlgorithm(Enum):
+    """Algorithm to use (relates to the low-level implementation)."""
```

Comment:
is the name algorithm correct here? Do you mean the algorithm used by Python and numba?